test/cases/reserved-cpus-kubelet-arg.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+echo "-> Should not set systemReservedCgroup and kubeReservedCgroup when --reserved-cpus is set with containerd"
+exit_code=0
+export KUBELET_VERSION=v1.24.15-eks-ba74326
+runtime="containerd"
+/etc/eks/bootstrap.sh \
+  --b64-cluster-ca dGVzdA== \
+  --apiserver-endpoint http://my-api-endpoint \
+  --kubelet-extra-args '--node-labels=cnf=cnf1 --reserved-cpus=0-3 --cpu-manager-policy=static' \
+  test || exit_code=$?
+
+if [[ ${exit_code} -ne 0 ]]; then
+  echo "❌ Test Failed: expected a non-zero exit code but got '${exit_code}'"
+  exit 1
+fi
+
+retCode=$(
+  grep -q systemReservedCgroup /etc/kubernetes/kubelet/kubelet-config.json
+  echo $?
+)
+if [[ ${retCode} -eq 0 ]]; then
```

Comment:
You can write this sort of thing like:
```suggestion
if grep -q systemReservedCgroup /etc/kubernetes/kubelet/kubelet-config.json; then
```

I would also use a variable instead of repeating the kubelet config path several times.

Reply:
@cartermckinnon committed changes and resolved all the comments